packages/dart/npt_flutter/lib/features/onboarding/util/profile_progress_listener.dart

```diff
@@ -5,16 +5,20 @@ import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:npt_flutter/app.dart';
 import 'package:npt_flutter/features/profile_list/bloc/profile_list_bloc.dart';
 
+import '../../profile_list/cubit/sync_cubit.dart';
+
 class ProfileProgressListener extends SyncProgressListener {
   @override
-  void onSyncProgressEvent(SyncProgress syncProgress) {
+  void onSyncProgressEvent(SyncProgress syncProgress) async {
+    await App.navState.currentContext!.read<SyncCubit>().checkSync();
     final profileListBlock = App.navState.currentContext!.read<ProfileListBloc>();
 
     if (syncProgress.syncStatus == SyncStatus.success &&
         (profileListBlock.state is ProfileListLoaded &&
             (profileListBlock.state as ProfileListLoaded).profiles.isEmpty)) {
       profileListBlock.add(const ProfileListLoadEvent());
       log('ProfileProgressListener: ProfileListLoadEvent triggered to reload profiles');
+      await App.navState.currentContext!.read<SyncCubit>().checkSync();
```

Comment:
use `unawaited()` instead of `await` to let the job get handled asynchronously. Especially for the one at the top, it will ensure the profiles are loaded as quickly as possible.